test/lib/spec.index.js

```diff
@@ -216,32 +246,40 @@ describe('CountriesCachedModel', () => {
             instance._indexCountries.should.be.a('function');
         });
 
-        it('should set _overseasCountries to all countries except the UK', () => {
+        it('should set _overseasCountries to all countries except GB', () => {
             instance._indexCountries();
             instance._overseasCountries.should.deep.equal([
-                { id: 'FO', slug: 'foo' },
-                { id: 'BA', slug: 'bar' },
-                { id: 'NA', slug: 'narnia' }
+                { addressCountryFlag: true, countryCode: 'FO', countryNameSlug: 'foo' },
+                { addressCountryFlag: false, countryCode: 'BA', countryNameSlug: 'bar' },
+                { addressCountryFlag: null, countryCode: 'NA', countryNameSlug: 'narnia' }
+            ]);
+        });
+
+        it('should set _residenceCountries to all countries where addressCountryFlag is true', () => {
+            instance._indexCountries();
+            instance._residenceCountries.should.deep.equal([
+                { addressCountryFlag: true, countryCode: 'GB', countryNameSlug: 'united-kingdom' },
+                { addressCountryFlag: true, countryCode: 'FO', countryNameSlug: 'foo' }
```

Comment:
There should also be a test for resident overseas countries indexing

Reply:
done